### ISSUE TITLE
decide how we want to expose CAs for OIDC in WCs

### ISSUE THREAD

**Issue by anvddriesch** [giantswarm/roadmap#1078]
Towards https://github.com/giantswarm/roadmap/issues/489
In order to implement SSO for WCs into `kubectl-gs login` we need a way to fetch the needed CA as well as potentially other information prior to authentication.
We want this to work without having to log into the MC first, therefore this information has to be exposed.

In the MC case we need to provide a URL that the clusters `athena` URL can be derived from (api or happa currently) to `kgs`. The `athena` is then queried for the following information.

- kubernetes api url
- internal kubernetes api url (derived from the above)
- oidc issuer url
- provider
- cluster name
- CA cert

### Options

1. Install `athena` on WCs
  pro: makes WC  more similar to MC, we could go towards offering athena as a managed app
  con: overkill just for this use case
2. Introduce a new lightweight service to expose only this information
  pro: lightweight, could be included in dex chart and potentially replace `k8s-authenticator`
  con: we need to decide on something

**Comment by anvddriesch** [comment]
I will try to install athena in a WC and see if the login function for MCs works.

**Comment by anvddriesch** [comment]
I have installed `athena` on one of our WCs where we already have `dex`. It is possible to log into this WC now via `kubectl gs login $wc_api_url` without making any changes to `kubectl-gs` itself. (It is being treated the same as our MCs)

Of course it was needed to provide `values.yaml` for athena. Since it is currently in the `controlplane catalog` and not a managed app there could be a potential to simplify this for WC usage.
`values.yaml` for the `dex-app` also needed some changes. Most notably I could only enable `kgs login` to work with the `giantswarm` connector after setting `isManagementCluster: true`. I believe we can improve configuration there as well.
I also had to add a `clusterrolebinding` on the WC for the `giantswarm` connector group to have admin rights. (No permissions were granted initially).
